examples/hello_world/client/nodejs/index.js

```diff
@@ -53,31 +46,34 @@ async function main() {
     return metaData;
   }
 
-  const helloWorldProto = grpc.loadPackageDefinition(helloWorldDefinition).oak
-    .examples.hello_world;
+  const helloWorldService = protos.lookupService(
+    'oak.examples.hello_world.HelloWorld'
+  );
   const credentials = grpc.credentials.createSsl(fs.readFileSync(CERT_PATH));
 
-  const client = new helloWorldProto.HelloWorld('localhost:8080', credentials);
+  const grpcClient = new grpc.Client('localhost:8080', credentials);
 
-  // For documentation on client calls see the `@grpc/grpc-js` documentation:
-  // https://grpc.github.io/grpc/node/grpc.Client.html#~CallProperties
-  client.sayHello(
-    // The arguments passed to the gRPC service. Corresponds to the
-    // `HelloRequest` message type in hello_world.proto file.
-    { greeting: 'Node.js' },
-    // The metadata for this gRPC call.
-    getGrpcMetadata(),
-    // Callback invoked with the response.
-    (error, response) => {
-      if (error) {
-        console.error(error);
-        process.exit(1);
-      } else {
-        console.log(response.reply);
-        process.exit(0);
-      }
+  const HelloWorld = helloWorldService.create(
+    (method, requestData, callback) => {
+      grpcClient.makeUnaryRequest(
+        `/oak.examples.hello_world.HelloWorld/${method.name}`,
+        (arg) => arg,
+        (arg) => arg,
+        requestData,
+        getGrpcMetadata(),
+        callback
+      );
```

Comment:
I'm not sure I understand what's going on here, is this the normal way of making a gRPC call? The `helloWorldService.create` sounds a bit awkward, and also not sure what the `(arg) => arg` things are.

Reply:
Good point, I added comments to better explain what is going on here.